### ISSUE TITLE
Experiment: use js-ipfs as a fallback for uploading content when IPFS API is down

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/ipfs-companion#304]
Here is an idea (well, a rough brain-dump actually – sorry!)

While we can't use embedded js-ipfs as HTTP gateway replacement at the moment  (elaborated on this in #248 and https://github.com/ipfs/in-web-browsers/pull/60),  we could use it as a **fallback solution for uploading content**. 

Consider a scenario:
_IPFS API is down and user tries to upload content_ (via context menu or using Quick Upload).

It should be possible for the browser extension to:
- open a new (pinned?) tab and spawn an ephemeral js-ipfs node in it
  - display basic status in the tab (eg. connecting to swarm etc)
    - if js-ipfs can't connect to any peers, display a sad face and inform user what happened and finish work
    - if connected, add content to the node

- when content is hashed and node is connected to the swarm, open resource CID using URL at a  preferred public gateway (this happens in a new tab)
  - public gateway downloads content from ephemeral node
  - when a tab with resource at public gateway finishes loading, we can:
    -  close the tab with ephemeral node,
    -  or just update status (display a happy face) inform user that CID is preloaded at a public gateway, but will disappear if it is not pinned anywhere soon etc

This would make browser extension more useful  for  people that are unable to run 3rd party daemon.

**Comment by lidel** [comment]
A related work is done in PR #310 👀

**Comment by lidel** [comment]
Delivered in #310 and #381. 

**tl;dr** Users can now toggle between embedded and external node, so it is always possible to upload content, even when user has  no `go-ipfs` and is running extension under Firefox/Chrome.
